lib/runtime/wazero/imports.go

```diff
@@ -2117,25 +2124,30 @@ func ext_storage_clear_prefix_version_2(ctx context.Context, m api.Module, prefi
 		limitPtr = &maxLimit
 	}
 
-	numRemoved, all, err := storage.ClearPrefixLimit(prefix, *limitPtr)
-	if err != nil {
-		logger.Errorf("failed to clear prefix limit: %s", err)
-		panic(err)
-	}
+	if !bytes.HasPrefix(prefix, childStorageKeyPrefix) {
+		numRemoved, all, err := storage.ClearPrefixLimit(prefix, *limitPtr)
+		if err != nil {
+			logger.Errorf("failed to clear prefix limit: %s", err)
+			panic(err)
+		}
 
-	encBytes, err := toKillStorageResultEnum(all, numRemoved)
-	if err != nil {
-		logger.Errorf("failed to allocate memory: %s", err)
-		panic(err)
-	}
+		encBytes, err := toKillStorageResultEnum(all, numRemoved)
+		if err != nil {
+			logger.Errorf("failed to allocate memory: %s", err)
+			panic(err)
+		}
 
-	valueSpan, err := write(m, rtCtx.Allocator, encBytes)
-	if err != nil {
-		logger.Errorf("failed to allocate: %s", err)
-		panic(err)
-	}
+		valueSpan, err := write(m, rtCtx.Allocator, encBytes)
+		if err != nil {
+			logger.Errorf("failed to allocate: %s", err)
+			panic(err)
+		}
 
-	return valueSpan
+		return valueSpan
+	} else {
+		logger.Warnf("cannot clear prefix that is part of or contains a child storage key")
+	}
+	return 0
```

Comment:
Wanted to double-check that we want to ignore this case and not panic. Also here wanted to make sure 0 was the correct value to return